x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_comments.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useState, useCallback, useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import {
+  EuiTextArea,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiAvatar,
+  EuiAccordion,
+  EuiCommentList,
+  EuiCommentProps,
+  EuiText,
+} from '@elastic/eui';
+import { Comment } from '../../../lists_plugin_deps';
+import * as i18n from './translations';
+import { useCurrentUser } from '../../lib/kibana';
+import { getFormattedComments } from './helpers';
+
+interface AddExceptionCommentsProps {
+  exceptionItemComments?: Comment[];
+  newCommentValue: string;
+  newCommentOnChange: (value: string) => void;
+}
+
+const commentAccordionButtonClassname = 'exceptionCommentAccordionButton';
+
+const MyAvatar = styled(EuiAvatar)`
+  ${({ theme }) => css`
+    margin-right: ${theme.eui.paddingSizes.m};
+  `}
+`;
+
+const CommentAccordion = styled(EuiAccordion)`
+  ${({ theme }) => css`
+    .${commentAccordionButtonClassname} {
+      color: ${theme.eui.euiColorPrimary};
+      padding: ${theme.eui.paddingSizes.m} 0;
+    }
+  `}
+`;
+
+export const AddExceptionComments = memo(function AddExceptionComments({
+  exceptionItemComments,
+  newCommentValue,
+  newCommentOnChange,
+}: AddExceptionCommentsProps) {
+  const [shouldShowComments, setShouldShowComments] = useState(false);
+  const currentUser = useCurrentUser();
+
+  const handleOnChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      newCommentOnChange(event.target.value);
+    },
+    [newCommentOnChange]
+  );
+
+  const handleTriggerOnClick = useCallback((isOpen: boolean) => {
+    setShouldShowComments(isOpen);
+  }, []);
+
+  const shouldShowAccordion: boolean = useMemo(() => {
+    return (
+      exceptionItemComments !== undefined &&
+      exceptionItemComments !== null &&
+      exceptionItemComments.length > 0
+    );
+  }, [exceptionItemComments]);
+
+  const commentsAccordionTitle = useMemo(() => {
+    if (exceptionItemComments && exceptionItemComments.length > 0) {
```

Comment:
Can you use the `shouldShowAccordion` here since it does the same check?

Reply:
ts isn't able to parse the logic from that properly and still lists `exceptionItemComments` below as potentially undefined